src/binder/bind/bind_copy.cpp

```diff
@@ -16,25 +16,59 @@ std::unique_ptr<BoundStatement> Binder::bindCopy(const Statement& statement) {
     validateTableExist(catalog, tableName);
     auto tableID = catalogContent->getTableID(tableName);
     auto csvReaderConfig = bindParsingOptions(copyCSV.getParsingOptions());
-    auto filePaths = bindFilePaths(copyCSV.getFilePaths());
-    auto fileType = bindFileType(filePaths);
+    auto boundFilePaths = bindFilePaths(copyCSV.getFilePaths());
+    auto fileType = bindFileType(boundFilePaths);
+    std::vector<std::string> filePaths;
+    std::unordered_map<std::string, std::string> propertyToNpyMap;
+    if (fileType == CopyDescription::FileType::NPY) {
```

Comment:
If the filetype is NPY, the propertyToNpyMap will be set and filePaths will be empty.
If the filetype is not NPY, the propertyToNpyMap will be empty and filePaths will be used.
I think it is better to extract a base class CopyDescription, and have 3 derived CopyDescription class:
CopyCSVDescription, CopyParquetDescription, CopyNpyDescription. We should move the fields which are only used by a derived class from the base class.

Reply:
I tried to refactor this, but since `CopyCSVDescription` is passed through multiple layers, this refactor would require changing all the values along the line into pointer. Otherwise it cannot be casted back to the original type in the last layer. After discussing with @andyfengHKU, we decided not to refactor this for now.